packages/core/src/request-builder/header-builder/csrf-token-header.ts

```diff
@@ -13,45 +12,44 @@ const logger = createLogger({
 });
 
 /**
- * Get CSRF token and cookies for a request with destination related headers.
- * @param request The request to get CSRF headers for.
- * @param headers Destination related headers to include in the CSRF fetch request.
+ * Get CSRF token and cookies for a destination and request configuration. The CSRF token and cookies will be retrieved based on the url of the destination and custom configuration given by the `requestConfig`.
+ * @param destination The destination to get the headers from
+ * @param requestConfig An http request configuration containing additional information about the request, like url or headers
  * @returns A promise to an object containing the CSRF related headers
  */
-export async function buildCsrfHeaders<RequestT extends ODataRequestConfig>(
-  request: ODataRequest<RequestT>,
-  headers: MapType<string>
+export async function buildCsrfHeaders<T extends HttpRequestConfig>(
+  destination: Destination | DestinationNameAndJwt,
+  requestConfig: Partial<T>
 ): Promise<MapType<string>> {
-  const csrfHeaders = await makeCsrfRequest(request, headers);
+  const csrfHeaders = await makeCsrfRequest(destination, requestConfig);
   validateCsrfTokenResponse(csrfHeaders);
   return filterNullishValues({
     ...getHeader('x-csrf-token', csrfHeaders),
     cookie: buildCookieHeaderValue(getHeaderValue('set-cookie', csrfHeaders))
   });
 }
 
-function makeCsrfRequest<RequestT extends ODataRequestConfig>(
-  request: ODataRequest<RequestT>,
-  csrfRequestHeaders: MapType<string>
+function makeCsrfRequest<T extends HttpRequestConfig>(
+  destination: Destination | DestinationNameAndJwt,
+  requestConfig: Partial<T>
 ): Promise<MapType<any>> {
-  if (!request.destination) {
-    throw Error('The request destination is undefined.');
-  }
-
-  const axiosConfig = {
+  const fetchHeader = !getHeaderValue(
+    'x-csrf-token',
+    requestConfig.headers
+  ) && { 'x-csrf-token': 'Fetch' };
+  const axiosConfig: HttpRequestConfig = {
+    method: 'get',
+    ...requestConfig,
     headers: {
-      ...csrfRequestHeaders,
-      'x-csrf-token': 'Fetch',
-      ...getAxiosConfigWithDefaults()
+      ...fetchHeader,
+      ...requestConfig.headers
     },
-    url: request.serviceUrl(),
-    ...getAgentConfig(request.destination)
+    url: requestConfig.url
```

Comment:
In the JS-DOC it is mentioned that the URL from the destination is used. But here it looks like the request config URL is used to fetch the csrf token.

Reply:
Unfortunately this is a little confusing. The url in the requestConfig will be appended to a baseUrl that we build from the destination given that the url is relative. If it is a absolute url it overwrites the baseUrl.

Reply:
I improved the documentation a little.

Reply:
ah. I was not aware of this but was expecting something like this, which was part of my next comment on the relative service path.